zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/internal/fetcher/FsCachingFetcher.kt

```diff
@@ -34,9 +34,9 @@ internal class FsCachingFetcher(
     nowEpochMs: Long,
     baseUrl: String?,
     url: String,
-  ): ByteString? {
+  ): ByteString {
     return cache.getOrPut(applicationName, sha256, nowEpochMs) {
-      delegate.fetch(applicationName, eventListener, id, sha256, nowEpochMs, baseUrl, url)
+      delegate.fetch(applicationName, eventListener, id, sha256, nowEpochMs, baseUrl, url)!!
```

Comment:
The concrete implementation of `Delegate` is an HttpFetcher that never returns null. (Though it does throw exceptions).

Reply:
Can we move that invariant up to the property type? Then presumably its covariant override will eliminate the need for this double bang.

Reply:
https://github.com/cashapp/zipline/pull/1375